### PR TITLE
Added Handling for Instruction Operand Memory Offsets [bx+di]

### LINKED ISSUE #75: [ELWIC] Infinity Complex - Module crashes on shutdown
**Module Information**
* Identifier: ELWIC
* Module Name: Infinity Complex
* Version: 6.04

**Describe the bug**
Module crashes on shutdown

**To Reproduce**
1. Add module to moduleConfig.json
2. Start MBBSEmu
3. CTRL-C and shutdown MBBSEmu
4. Module crashes on shutdown routine

**Expected behavior**
Module shuts down normally

**Screenshots**
```
2020-09-01 20:12:48.6764 Info MBBSEmu.HostProcess.MbbsHost.<Shutdown>b__18_0 Calling shutdown routine on module ELWIC
2020-09-01 20:12:48.6888 Warn MBBSEmu.Btrieve.BtrieveFileProcessor.SeekByKey Adjusting Query Key Size, Data Size 7 differs from Defined Key Size 25
2020-09-01 20:12:48.6888 Warn MBBSEmu.Btrieve.BtrieveFileProcessor.SeekByKey Adjusting Query Key Size, Data Size 8 differs from Defined Key Size 25
Unhandled exception. System.Exception: Unknown GetOperandOffset MemoryBase
   at MBBSEmu.CPU.CpuCore.GetOperandOffset(OpKind opKind)
   at MBBSEmu.CPU.CpuCore.GetOperandValueUInt16(OpKind opKind, EnumOperandType operandType)
   at MBBSEmu.CPU.CpuCore.Tick()
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.CallModuleRoutine(String routine, Action`1 preRunCallback, UInt16 channel)
   at MBBSEmu.HostProcess.MbbsHost.Shutdown()
   at MBBSEmu.HostProcess.MbbsHost.WorkerThread()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadHelper.ThreadStart()
```

**Software Information:**
- OS: Windows 2019 Server Standard
- Telnet Client: N/A

**API Report**
{"UniqueIdentifier":"ELWIC","File":"ELWIC.DLL","Imports":{"DOSCALLS":[89],"PHAPI":[16],"MAJORBBS":[441,574,326,544,65,437,455,571,566,543,158,492,589,331,550,559,601,357,534,757,444,520,456,474,629,463,628,476,403,650,317,485,624,573,335,522,401,229,475,462,225,637,205,565,510,467,420,350,511,486,351,117,162,167,713,230,560,516,433,654,665,113,477,578,584,505,621,313,53,226,94,11,408,77,402,334,718,579,604,582,580,366,377,659,581,603],"GALGSBL":[40,37,59,21,7,72,36]}}

### PR DESCRIPTION
**Pull request by enusbaum**
Fixes #75

### CODE REVIEW

File: MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            Reset();
+            CreateDataSegment(new ReadOnlySpan<byte>(), 2);
+            mbbsEmuMemoryCore.SetWord(2, (ushort)(initialDI + initialBX), memoryInitialValue);
+            mbbsEmuCpuCore.Registers.DS = 2;
+            mbbsEmuCpuCore.Registers.BX = initialBX;
+            mbbsEmuCpuCore.Registers.DI = initialDI;
+
+            var instructions = new Assembler(16);
+            instructions.add(__word_ptr[bx + di], valueToAdd);
```
Comment (paladine) [inline MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs:26]: I'm curious how this compiles, but I'll just assume Iced.Intel is doing some magic here

File: MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+namespace MBBSEmu.Tests.CPU
+{
+    public class GetOperandOffset_Tests : CpuTestBase
+    {
+
+        [Theory]
+        [InlineData(0, 1, 1, 0, 1)]
+        [InlineData(1, 1, 2, 0, 0)]
+        [InlineData(0xFFFF, 1, 0, 1, 0)]
+        [InlineData(0xFFFE, 1, 0xFFFF, 1, 1)]
```
Comment (paladine) [inline MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs:15]: what happens if [BX + DI] overflows 65535, does the processor access the lower value or does it invoke an Overflow interrupt?

Comment (enusbaum) [reply MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs:15]: i'm not sure -- but that'd be an error in the compiler if that case were to happen as no segment in an NE format DLL/EXE is > 0xFFFF bytes.

File: MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs (hunk 1 above)
Comment (enusbaum) [reply MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs:26]: I pulled the example from [their documentation](https://github.com/0xd4d/iced/blob/master/src/csharp/Intel/README.md) on how to do it. Verified in the CPU core, the instruction is correct 👍

File: MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs (hunk 2 above)
Comment (paladine) [reply MBBSEmu.Tests/CPU/GetOperandOffset_Tests.cs:15]: I'm pretty sure MajorMUD relies on overflowing memory references as a copy protection mechanism.

Just kidding.

**Pull request merged**